src/ui/server/controllers/insurance/your-buyer/outstanding-or-overdue-payments/index.ts

```diff
@@ -0,0 +1,195 @@
+import generateValidationErrors from './validation';
+import { Request, Response, Currency } from '../../../../../types';
+import constructPayload from '../../../../helpers/construct-payload';
+import mapApplicationToFormFields from '../../../../helpers/mappings/map-application-to-form-fields';
+import mapAndSave from '../map-and-save/buyer-trading-history';
+import isChangeRoute from '../../../../helpers/is-change-route';
+import isCheckAndChangeRoute from '../../../../helpers/is-check-and-change-route';
+import getCurrencyByCode from '../../../../helpers/get-currency-by-code';
+import api from '../../../../api';
+
+const {
+  INSURANCE_ROOT,
+  YOUR_BUYER: { OUTSTANDING_OR_OVERDUE_PAYMENTS_SAVE_AND_BACK: SAVE_AND_BACK, CHECK_YOUR_ANSWERS, FAILED_TO_PAY },
+  CHECK_YOUR_ANSWERS: { YOUR_BUYER: CHECK_AND_CHANGE_ROUTE },
+  PROBLEM_WITH_SERVICE,
+} = INSURANCE_ROUTES;
+
+const {
+  CURRENCY: { CURRENCY_CODE },
+} = INSURANCE_FIELD_IDS;
+
+const { TOTAL_OUTSTANDING_PAYMENTS, TOTAL_AMOUNT_OVERDUE } = YOUR_BUYER_FIELD_IDS;
+
+export const FIELD_IDS = [TOTAL_OUTSTANDING_PAYMENTS, TOTAL_AMOUNT_OVERDUE];
+
+export const TEMPLATE = TEMPLATES.INSURANCE.YOUR_BUYER.OUTSTANDING_OR_OVERDUE_PAYMENTS;
+
+export const PAGE_CONTENT_STRINGS = PAGES.INSURANCE.YOUR_BUYER.OUTSTANDING_OR_OVERDUE_PAYMENTS;
+
+/**
+ * pageVariables for outstanding-or-overdue page
+ * @param {Number} referenceNumber: Application reference number
+ * @param {Array<Currency>} currencies: Array of currencies
+ * @param {String} currencyCode: Provided currency code
+ * @param {Boolean} changeRoute: req.originalUrl is a change route
+ * @param {Boolean} checkAndChangeRoute: req.originalUrl is a check-and-change route
+ * @returns {Object} pageVariables
+ */
+export const pageVariables = (referenceNumber: number, currencies: Array<Currency>, currencyCode: string) => {
+  const currency = getCurrencyByCode(currencies, currencyCode);
+
+  return {
+    FIELDS: {
+      TOTAL_OUTSTANDING_PAYMENTS: {
+        ID: TOTAL_OUTSTANDING_PAYMENTS,
+        ...FIELDS[TOTAL_OUTSTANDING_PAYMENTS],
+      },
+      TOTAL_AMOUNT_OVERDUE: {
+        ID: TOTAL_AMOUNT_OVERDUE,
+        ...FIELDS[TOTAL_AMOUNT_OVERDUE],
+      },
+    },
+    PAGE_CONTENT_STRINGS,
+    SAVE_AND_BACK_URL: `${INSURANCE_ROOT}/${referenceNumber}${SAVE_AND_BACK}`,
+    CURRENCY_PREFIX_SYMBOL: currency.symbol,
+  };
+};
+
+/**
+ * get
+ * Render the outstanding or overdue payments page
+ * @param {Express.Request} Express request
+ * @param {Express.Response} Express response
+ * @returns {Express.Response.render} Outstanding or overdue paymentspage
+ */
+export const get = async (req: Request, res: Response) => {
+  try {
+    const { application } = res.locals;
+
+    if (!application) {
+      return res.redirect(PROBLEM_WITH_SERVICE);
+    }
+
+    const {
+      referenceNumber,
+      buyer: { buyerTradingHistory },
+    } = application;
+
+    const { supportedCurrencies } = await api.keystone.APIM.getCurrencies();
```

Comment:
Can we add a redirect to `PROBLEM_WITH_SERVICE` if `supportedCurrencies` is not populated? We have this in other places